students/JeroenSweerts/Lesson3/mailroom.py

```diff
@@ -15,12 +15,14 @@ def thankyou():
     the selected user.
     Finally, use string formatting to compose an email thanking the donor for their
     generous donation. Print the email to the terminal and return to the original
-    prompt.'''
-    name = input('Please enter a name or type "list" to see a list of names > ')
-    while name == 'list':
-        print(donors)
+    prompt."""
+    while True:
         name = input('Please enter a name or type "list" to see a list of names > ')
-    if not name in donors:
+        if name == 'list':
+            print(donors)
+        else:
+            break
+    if name not in donors:
         donors.append(name)
         donations.append([])
     donation = input('Please enter a donation amount > ')
```

Comment:
If you convert and store all new donations as `int`s here, then you won't need to call `int()` on every donation where you generate the report below. And I would suggest using `float`, since we're dealing with money.